### PR TITLE
fix: client script added for tax row and tax breakup fix

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: https://bloomstack.com/desk#Form/Task/TASK-2020-01913
TASK ID: https://bloomstack.com/desk#Form/Task/TASK-2020-01527

1. fixed show only cultivation tax row which has amount
2. fixed tax breakup
Baldwin-Wells-PUR-ORD-2020-00060

### CODE REVIEW

File: erpnext/compliance/taxes.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ def calculate_cannabis_tax(doc):
 		# calculate cultivation tax for buying cycle
 		cultivation_taxes = calculate_cultivation_tax(doc)
 		for account, tax in cultivation_taxes.items():
+			if not tax:
+				continue
```
Comment (Alchez) [inline erpnext/compliance/taxes.py:27]: This won't work, I mentioned this in the other PR too. If you suddenly go from taxed to non-taxed (in any way), your old tax row won't get deleted. The deletion needs to happen post calculating the taxes and setting the rows.

File: erpnext/controllers/taxes_and_totals.py [hunk 2]
```diff
@@ -631,7 +631,6 @@ def get_itemised_tax_breakup_html(doc):
 
 	# get tax breakup data
 	itemised_tax, itemised_taxable_amount = get_itemised_tax_breakup_data(doc)
-
```
Comment (Alchez) [inline erpnext/controllers/taxes_and_totals.py:634]: Remove this change too if there's nothing else

File: erpnext/public/js/controllers/buying.js [hunk 3]
```diff
@@ -224,6 +225,22 @@ erpnext.buying.BuyingController = erpnext.TransactionController.extend({
...
+
+	item_code: function(doc, cdt, cdn) {
+		this._super(doc, cdt, cdn);
+		if (this.frm.doc.total) {
+			this.set_and_update_cultivation_tax();
+		}
+	},
+
+	rate: function(doc, cdt, cdn) {
+		this.set_and_update_cultivation_tax();
```
Comment (Alchez) [inline erpnext/public/js/controllers/buying.js:240]: `_super` on all the triggers

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 4]
```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
...
+		if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+			if (!this.frm.doc.license) {
+				return;
+			}
+		}
+
+		const me = this;
+		frappe.db.get_value("Compliance Info", { "name": this.frm.doc.license }, "license_for", (r) => {
+			if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+				console.log("test");
```
Comment (Alchez) [inline erpnext/public/js/controllers/taxes_and_totals.js]: Remove `console.log`

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 5]
```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 				}
 			});
 		})
+	},
+	set_and_update_cultivation_tax: function () {
+		if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+			if (!this.frm.doc.license) {
```
Comment (Alchez) [inline erpnext/public/js/controllers/taxes_and_totals.js]: This check should happen in the selling controller instead, before this function is called

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 6]
```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
...
+				return;
+			}
+		}
+
+		const me = this;
+		frappe.db.get_value("Compliance Info", { "name": this.frm.doc.license }, "license_for", (r) => {
+			if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+				console.log("test");
+				if (!r || r.license_for !== "Distributor") {
```
Comment (Alchez) [inline erpnext/public/js/controllers/taxes_and_totals.js:769]: The `if (!r) return` check should happen for ALL doctypes

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 7]
```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
...
+			frappe.call({
+				method: "erpnext.compliance.taxes.set_cultivation_tax",
+				args: {
+					doc: this.frm.doc
+				},
+				callback: (r) => {
+					if (r.message) {
+						let taxes = me.frm.doc.taxes;
+						let excise_tax_row = r.message;
+						console.log("excise_tax_row", excise_tax_row);
```
Comment (Alchez) [inline erpnext/public/js/controllers/taxes_and_totals.js]: Remove `console.log`

File: erpnext/public/js/controllers/taxes_and_totals.js [hunk 8]
```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
...
+						if (excise_tax_row.tax_amount > 0) {
```
Comment (Alchez) [inline erpnext/public/js/controllers/taxes_and_totals.js]: Is this the same as the excise tax logic for setting tax rows? Maybe we should commonify it

File: erpnext/selling/sales_common.js [hunk 9]
```diff
@@ -486,21 +487,25 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
...
 	},
 
 	rate: function(doc, cdt, cdn) {
 		this.set_and_update_excise_tax();
+		this.set_and_update_cultivation_tax();
 	},
 
 	items_remove: function(doc, cdt, cdn) {
 		this.set_and_update_excise_tax();
+		this.set_and_update_cultivation_tax();
```
Comment (Alchez) [inline erpnext/selling/sales_common.js:508]: We should include these function calls in `calculate_taxes_and_totals`, these are just too many calls to do individually

**Review by Alchez — CHANGES_REQUESTED** [review]

File: erpnext/compliance/taxes.py (hunk 1 above)
Comment (neilLasrado) [reply erpnext/compliance/taxes.py:27]: @vishdha please fix this

**Review by neilLasrado — CHANGES_REQUESTED** [review]

**Comment by neilLasrado** [comment]
@vishdha whats happening with this PR?

**Pull request closed without merging**